PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,41 @@
+#Azure SDK for Java Pull Request Template Guide
```

Comment:
Could you use the Azure SDK For .NET PULL_REQUEST_TEMPLATE.md as a template for the Java repo?

https://github.com/Azure/azure-sdk-for-net/blob/master/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.md

The embedded links would need to be updated to point to resources in our repo, and those resources would also need to be added.